content.js

```diff
@@ -2,15 +2,17 @@ const logId = 'sticky-pinned-tabs [extension]';
 
 // TODO: observe dom mutation, maybe?
 
-function fixAllLinkTargets() {
+function fixAllLinkTargets(stickDomain) {
   console.log(
     `${logId}: ${
       window.location.href
     } Changing all links to open in target=_blank`,
   );
   for (const link of window.document.getElementsByTagName('a')) {
     if (link.href !== '' && link.href.indexOf('#') !== 0) {
-      link.target = '_blank';
+      if (!stickDomain || (stickDomain && ! new RegExp('/' + window.location.hostname, 'i').test(link.href))){
```

Comment:
Could you use [`link.origin`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement) instead of `href`? Here are [some details](https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/origin) about it.